java/configuration/src/main/java/sleeper/configuration/properties/table/TableProperty.java

```diff
@@ -66,51 +70,63 @@ public interface TableProperty extends SleeperProperty {
                     "When this is changed, existing files will retain their encryption status. Further compactions may " +
                     "apply the new encryption status for that data.\n" +
                     "See also: https://docs.aws.amazon.com/AmazonS3/latest/userguide/default-bucket-encryption.html")
-            .runCDKDeployWhenChanged(true).build();
+            .propertyGroup(TablePropertyGroup.DATA_DEFINITION)
+            .runCDKDeployWhenChanged(true)
+            .build();
     TableProperty ROW_GROUP_SIZE = named("sleeper.table.rowgroup.size")
             .defaultProperty(DEFAULT_ROW_GROUP_SIZE)
             .description("The size of the row group in the Parquet files - defaults to the value in the instance properties.")
+            .propertyGroup(TablePropertyGroup.DATA_DEFINITION)
             .build();
     TableProperty PAGE_SIZE = named("sleeper.table.page.size")
             .defaultProperty(DEFAULT_PAGE_SIZE)
             .description("The size of the page in the Parquet files - defaults to the value in the instance properties.")
+            .propertyGroup(TablePropertyGroup.DATA_DEFINITION)
             .build();
     TableProperty S3A_READAHEAD_RANGE = named("sleeper.table.fs.s3a.readahead.range")
             .defaultProperty(DEFAULT_S3A_READAHEAD_RANGE)
             .description("The S3 readahead range - defaults to the value in the instance properties.")
+            .propertyGroup(TablePropertyGroup.DATA_DEFINITION)
             .build();
     TableProperty COMPRESSION_CODEC = named("sleeper.table.compression.codec")
             .defaultProperty(DEFAULT_COMPRESSION_CODEC)
             .description("The compression codec to use for this table. Defaults to the value in the instance properties.")
+            .propertyGroup(TablePropertyGroup.DATA_DEFINITION)
             .build();
```

Comment:
These S3 and Parquet settings (starting at `sleeper.table.encrypted`) could be under a separate "Data Storage" group. They don't seem to really fit in "Data Definition".